native/explorer/src/lazyframe.rs

```diff
@@ -7,7 +7,11 @@ pub mod io;
 
 #[rustler::nif(schedule = "DirtyCpu")]
 pub fn lf_collect(data: ExLazyFrame) -> Result<ExDataFrame, ExplorerError> {
-    Ok(ExDataFrame::new(data.resource.0.clone().collect()?))
+    let mut df = data.resource.0.clone().collect()?;
+
+    let normalized_df = crate::dataframe::normalize_numeric_dtypes(&mut df)?;
+
+    Ok(ExDataFrame::new(normalized_df))
```

Comment:
Doing this change here is too late because all operations in the lazy frame would then act on i32, u32 and so on. We also need to do them immediately after reading the file/binary.

Reply:
I see. I don't know yet how to apply this change for a LazyFrame, so I decided to revert and keep it just for the  eager frame. I think I'm going to open an issue for that and tackle after the release. WDYT?